File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
public:
    bool visited[201]; 
    void dfs(vector<vector<int>>& isConnected, int node){
        visited[node] = true; 
        for(int i = 0; i < isConnected[node].size(); i++){
            if(i == node) continue; 
            int next = isConnected[node][i];
            if(!visited[i] && next == 1){
                dfs(isConnected,i); 
            }
        }
    }
    int findCircleNum(vector<vector<int>>& isConnected) {
        int count = 0; 
        for(int i = 0; i < isConnected.size(); i++){
            if(!visited[i]){
                dfs(isConnected,i); 
                count++; 
            }
        }
        return count; 
    }
};

File: 0547-number-of-provinces/0547-number-of-provinces.java
class Solution {
    public void dfs(int[][] isConnected, boolean[] visited, int node){
        visited[node] = true; 

        for(int i = 0; i < isConnected[node].length; i++){
            if(isConnected[node][i] == 1 && i != node && !visited[i]){
                dfs(isConnected,visited,i); 
            }
        }
    }
    public int findCircleNum(int[][] isConnected) {
        int n = isConnected.length; 
        boolean[] visited = new boolean[n];
        int answer = 0; 

        for(int i = 0; i < n; i++){
            if(!visited[i]){
                answer++; 
                dfs(isConnected,visited,i); 
            }
        }

        return answer; 
    }
}